lib/mappers/route_mapper.rb

```diff
@@ -12,6 +12,8 @@ def initialize(api_key, gateway_class = GoogleMaps::Routes::Api)
     def calculate_route(origin, destination)
       data = @gateway.route_data(origin.id, destination.id)
       DataMapper.new(data, origin, destination).build_entity
+    rescue NoMethodError
```

Comment:
Do we need this because DataMapper.new returns null?

Reply:
For handling when `origin` or `destination` is null.

Reply:
when trying to call `.id` on null object it will raise an error